### DISCUSSION TITLE
Proposed way for onActivityResult (Capacitor 6)

### DISCUSSION THREAD

**Discussion by vladanneon** [ionic-team/capacitor#7708] [Q&A]
How would you implement onActivityResult inside a Capacitor 6 plugin?

Docs mention that the preferred way is to use [startActivityForResult](https://capacitorjs.com/docs/plugins/android#intents-with-results) but SDK that I want to integrate mentions [onActivityResult](https://github.com/idnow/de.idnow.android?tab=readme-ov-file#usage)
If you take a glance at the SDK source code, you could see that the SDK itself calls: `this.callingActivity.startActivityForResult(this.startIdIntent, 2);`
during the 
`IDnowSDK.getInstance().start(IDnowSDK.getTransactionToken());`

How would you capture ActivityResult and return the result from the plugin back to the app?

**Comment by jcesarmobile** [comment]
You can check Capacitor 2.x docs, that should still work despite it's deprecated:
https://capacitorjs.com/docs/v2/plugins/android#intents-with-results

But use `@CapacitorPlugin` annotation instead of `@NativePlugin` annotation as that's deprecated too.
